src/sqlancer/QueryAdapter.java

```diff
@@ -61,9 +61,9 @@ public String getQueryString() {
     public boolean execute(GlobalState<?, ?> globalState, String... fills) throws SQLException {
         Statement s;
         if (fills.length > 0) {
-            s = globalState.getConnection().prepareStatement(getQueryString());
-            for (int i = 0; i < fills.length; i++) {
-                ((PreparedStatement) s).setString(i + 1, fills[i]);
+            s = globalState.getConnection().prepareStatement(fills[0]);
```

Comment:
I might have forgotten, but was there a specific reason why the template string is passed together with the fills, and not as part of a separate argument? I think packing both together is confusing (also considering that the name is misleading, if not only fills are passed).

Reply:
I had them separately beforehand, but I recall you wanted me to include the template in the query string, which resulted in the logs to not include the fills. Since I cannot have two variable length arguments, adding the template argument would require this argument to be used in all cases, including the cases where we don’t use prepared statements (but just the query). This would alter the behavior of the tool in large scale in my opinion.

Reply:
Sorry for getting confused about this, and thanks for the explanation. Do you think it would make sense to derive a `PreparedQuery` interface that extends `Query`, and a `PreparedQueryAdapter` that extends `QueryAdapter`? My concern is that the current approach is somewhat surprising, since, if I understand correctly, the query string is left empty and the two types of arguments are passed in the varargs argument.

Reply:
In the new approach in this PR, the query string isn‘t left empty. Instead it is set to be the filled-in version of the statement (but that is done explicitly, there are examples of that in the CitusProvider class in my other PR). This is so that when logging, the filled in version that is stored in the query is logged. I could implement a separate interface to do this implicitly as you suggested though.

Reply:
I think having this separated out as a separate (sub)class and interface would be better. Do you want to do that in a follow-up PR, so that we can get the Citus PR merged faster?

Reply:
Sounds good, but I wanted to clarify what you mean. Would you like to merge this PR and make the change we talked about later on, or would you like to cancel this PR? I will change the Citus implementation accordingly if it is the latter.

Reply:
I thought about first merging this PR, so you can do the changes after merging the Citus implementation.

Reply:
Sounds good to me, feel free to merge this PR!